Error - cannot be instantiated

Apache Maven 3.0.5 (r01de14724cdef164cd33c7c8c2fe155faf9602da; 2013-02-19 20:51:28+0700)
Java version: 1.7.0_55, vendor: Oracle Corporation
LDAP: Oracle Directory Enterprise Edition.

[ERROR] Failed to execute goal com.btmatthews.maven.plugins:ldap-maven-plugin:1.2.0:load (default-cli) on project ldapapp: Unable to parse configuration of mojo com.btmatthews.maven.plugins:ldap-maven-plugin:1.2.0:load for parameter ldif: Abstract class or interface 'com.btmatthews.maven.plugins.ldap.mojo.Ldif' cannot be instantiated -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.btmatthews.maven.plugins:ldap-maven-plugin:1.2.0:load (default-cli) on project ldapapp: Unable to parse configuration of mojo com.btmatthews.maven.plugins:ldap-maven-plugin:1.2.0:load for parameter ldif: Abstract class or interface 'com.btmatthews.maven.plugins.ldap.mojo.Ldif' cannot be instantiated


I'm not sure what is wrong, please help me to clarify.

--- a/ldap-maven-plugin/src/main/java/com/btmatthews/maven/plugins/ldap/mojo/AbstractSource.java
+++ b/ldap-maven-plugin/src/main/java/com/btmatthews/maven/plugins/ldap/mojo/AbstractSource.java
@@ -52,6 +52,13 @@ public abstract class AbstractSource implements Source {
     }
 
     /**
+    * Method call by plexus to valorise path after contruction of the class
+    */
+    public void set(String path){
+        this.path = path;
+    }
+
+    /**
      * Open the resource or file and return the input stream. If the path is prefixed with classpath: then it is treated
      * as a classpath resource. Otherwise, it is assumed to be a file system path.
      *

--- a/ldap-maven-plugin/src/main/java/com/btmatthews/maven/plugins/ldap/mojo/Dsml.java
+++ b/ldap-maven-plugin/src/main/java/com/btmatthews/maven/plugins/ldap/mojo/Dsml.java
@@ -32,4 +32,8 @@ public final class Dsml extends AbstractSource {
     public Dsml(final String path) {
         super(path);
     }
+
+    public Dsml(){
+    	super(null);
+    }
 }

--- a/ldap-maven-plugin/src/main/java/com/btmatthews/maven/plugins/ldap/mojo/Ldif.java
+++ b/ldap-maven-plugin/src/main/java/com/btmatthews/maven/plugins/ldap/mojo/Ldif.java
@@ -32,4 +32,8 @@ public final class Ldif extends AbstractSource {
     public Ldif(final String path) {
         super(path);
     }
+
+    public Ldif(){
+    	super(null);
+    }
 }

